routers/web/explore/code.go

```diff
@@ -37,9 +38,9 @@ func Code(ctx *context.Context) {
 	queryType := ctx.FormTrim("t")
 	isMatch := queryType == "match"
 
-	ctx.Data["Keyword"] = keyword
-	ctx.Data["Language"] = language
-	ctx.Data["queryType"] = queryType
+	ctx.Data["Keyword"] = url.PathEscape(keyword)
+	ctx.Data["Language"] = url.PathEscape(language)
+	ctx.Data["queryType"] = url.PathEscape(queryType)
```

Comment:
Old code should be kept. Only encode/escape the variables when it is really used. Otherwise there will be unavoidable double-escaping problems.

Reply:
ps: PathEscape is not right. It should be escaped by the template QueryEscape when they are put into the query parameter.

Reply:
Should be better now